tests/e2e/device-plugin_test.go

```diff
@@ -27,168 +27,154 @@ import (
+		// so we wait for all DaemonSets in the namespace rather than specific names
+		By("Waiting for all DaemonSets to be ready")
+		err = internal.WaitForAllDaemonSetsReady(ctx, clientSet, testNamespace.Name)
+		Expect(err).NotTo(HaveOccurred())
+	})
 
-		// check Collector objects
-		collectLogsFrom = defaultCollectorObjects
-		if *CollectLogsFrom != "" && *CollectLogsFrom != "default" {
-			collectLogsFrom = strings.Split(*CollectLogsFrom, ",")
+	AfterAll(func(ctx SpecContext) {
+		By("Uninstalling k8s-device-plugin Helm chart")
+		err := helmClient.UninstallReleaseByName(helmReleaseName)
+		if err != nil {
+			GinkgoWriter.Printf("Failed to uninstall helm release %s: %v\n", helmReleaseName, err)
 		}
+	})
 
-		BeforeAll(func(ctx context.Context) {
-			// Create clients for apiextensions and our CRD api
-			extClient = extclient.NewForConfigOrDie(f.ClientConfig())
-			helmReleaseName = "nvdp-e2e-test" + rand.String(5)
-
-			chartSpec = helm.ChartSpec{
-				ReleaseName:   helmReleaseName,
-				ChartName:     *HelmChart,
-				Namespace:     f.Namespace.Name,
-				Wait:          true,
-				Timeout:       1 * time.Minute,
-				ValuesOptions: values,
-				CleanupOnFail: true,
-			}
-
-			By("Installing k8s-device-plugin Helm chart")
-			_, err := f.HelmClient.InstallChart(ctx, &chartSpec, nil)
+	AfterEach(func(ctx SpecContext) {
+		// Run diagnostic collector if test failed
+		if CurrentSpecReport().Failed() {
+			var err error
+			diagnosticsCollector, err = diagnostics.New(
+				diagnostics.WithNamespace(testNamespace.Name),
+				diagnostics.WithArtifactDir(LogArtifactDir),
+				diagnostics.WithKubernetesClient(clientSet),
+				diagnostics.WithObjects(collectLogsFrom...),
+			)
 			Expect(err).NotTo(HaveOccurred())
-		})
 
-		JustBeforeEach(func(ctx context.Context) {
-		})
+			err = diagnosticsCollector.Collect(ctx)
+			Expect(err).NotTo(HaveOccurred())
+		}
+	})
 
-		AfterEach(func(ctx context.Context) {
-			// Run diagnostic collector if test failed
-			if CurrentSpecReport().Failed() {
-				var err error
-				diagnosticsCollector, err = diagnostics.New(
-					diagnostics.WithNamespace(f.Namespace.Name),
-					diagnostics.WithArtifactDir(*LogArtifactDir),
-					diagnostics.WithKubernetesClient(f.ClientSet),
-					diagnostics.WithObjects(collectLogsFrom...),
-				)
-				Expect(err).NotTo(HaveOccurred())
-
-				err = diagnosticsCollector.Collect(ctx)
-				Expect(err).NotTo(HaveOccurred())
-			}
-		})
+	When("When deploying k8s-device-plugin", Ordered, Label("serial"), func() {
+		It("it should create nvidia.com/gpu resource", Label("gpu-resource"), func(ctx SpecContext) {
+			nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(nodeList.Items)).ToNot(BeZero())
 
-		AfterAll(func(ctx context.Context) {
-			// Delete Helm release
-			err := f.HelmClient.UninstallReleaseByName(helmReleaseName)
+			// We pick one node
+			nodes, err := getNonControlPlaneNodes(ctx, clientSet)
```

Comment:
Why are we considering control plane nodes here? These only have GPU resources if they are the SAME node as a worker node.